### ISSUE TITLE
SCSS @use rule and ViewTransition Bug at reusable/common Astro Components

### ISSUE THREAD

**Issue by igor-ht** [withastro/astro#8453]
### Astro Info

```block
Astro                    v3.0.10
Node                     v18.17.1
System                   Windows (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             @astrojs/mdx, @astrojs/sitemap
```

### Describe the Bug

When navigating using ```ViewTransition``` through routes that share a common ```Astro Layout``` or ```Astro Component``` with a SCSS stylesheet that uses the **SCSS rule ```@use```**, the browser triggers this issue:
![My Blog - Google Chrome 9_7_2023 6_22_12 PM (2)](https://github.com/withastro/astro/assets/117295025/39f7cb5a-51f4-406a-93fc-5836ce66c33c)

When navigating without ```ViewTransition``` or when removing the SCSS ```@use``` rule from the reusable component only, there's no issue.

### What's the expected result?

Be able to use the SCSS ```@use``` rule at any common/reusable Astro component together with ```ViewTransition``` without triggering any issues.

### Link to Minimal Reproducible Example

https://stackblitz.com/github/igor-ht/my-blog

### Participation

- [X] I am willing to submit a pull request for this issue.

#### Edit:
Astro info fixed.

**Comment by igor-ht** [comment]
It's also possible to observe that the _ToggleTheme_ method works perfectly when refreshing the browser at any route and toggling between **light** and **dark**. The current theme is also passed through pages when navigating between them. But when navigating to any route (here is when the ```Verify stylesheet URLs``` issue appears)  and trying to toggle the theme, it won't trigger the method, but after refreshing the browser the theme can be toggled again.

**Comment by igor-ht** [comment]
After some tests I found out the problem was with importing a google font with an external URL in a SCSS file that was being used with the ```@use``` rule. When using ```ViewTransition``` to navigate through routes, the issue actually refers to the URL of google font, and not the stylesheet.

The solution I used was to follow the docs (https://docs.astro.build/en/guides/fonts/) and change to the @font-face method with a font file in the public folder.
